src/wp-includes/class-wp-query.php

```diff
@@ -3060,6 +3056,74 @@ public function get_posts() {
 		 */
 		$this->posts = apply_filters_ref_array( 'posts_pre_query', array( null, &$this ) );
 
+		/*
+		 * Ensure the ID database query is able to be cached.
+		 *
+		 * Random queries are expected to have unpredictable results and cannot be cached.
+		 */
+		$id_query_is_cacheable = ! preg_match( '/\\sRAND\\(\\d*\\)(\\s|$)/i', $orderby );
```

Comment:
A logic check here would be dandy. 

Escaping slashes in regex always confuses the heck out of me.